src/classes/action/link_action.vala

```diff
@@ -114,6 +114,33 @@ namespace pdfpc {
             );
         }
 
+        /**
+         * Launch an application, trying to cover all corner cases.
+         */
+        private void launch_for_uri(string uri) throws GLib.Error {
+            if (Uri.parse_scheme(uri) != null) {
+                AppInfo.launch_default_for_uri(uri, null);
```

Comment:
I just read over the docs and `launch_default_for_uri_async` looks more promising to me?

Reply:
Do you think it's reasonable for pdfpc to provide any type of GUI selector to the user during a presentation?

More generally, I don't quite understand the need for clickable URI links; it's against the pdfpc principles to show identical views on two screens...

Reply:
You never know what weird setups people have. And as far as I understand the "select application GUI" only shows up if you have no default.

I use sometimes clickable URI (mostly web pages, sometimes other files on my disk) in my presentations.

Reply:
Well, but then the web page appears on your screen, not where the people in the auditorium can see it?